src/Core/Tool/Authorizer/PostAuthorizer.php

```diff
@@ -114,7 +114,7 @@ public function isAllowed(Entity $entity, $privilege)
         }
 
         // Non-admin users are not allowed to create posts for forms that have restricted access.
-        if (in_array($privilege, ['create', 'update'])
+        if (in_array($privilege, ['create', 'update', 'lock'])
```

Comment:
check: lock is a permission?

Reply:
In this context, it's a privilege which is more like an action. There is a divide between these and permissions. The permissions operate as separate controls, this is more like a descriptor of the action being performed. It is possible to restrict what actions are possible.